Hyperion Crashsite

biscuitWizard opened the pull request:
Added the hyperion to the exoplanet feature generation list. The hyperion is a long-range explorer ship that crash landed. It features a R-UST propulsion system, but requires some additional work to get flight worthy. Or is a source of parts for The Torch.

Commit:
Added the hyperion to the exoplanet feature generation list. The hyperion is a long-range explorer ship that crash landed. It features a R-UST propulsion system, but requires some additional work to get flight worthy. Or is a source of parts for The Torch.

Commit:
Mapmerged hyperion.

comma commented on maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dm:
```diff
@@ -0,0 +1,60 @@
+	vessel_size = SHIP_SIZE_LARGE
+
+//////////////////////////
+// /datum/shuttle/autodock/overmap/swordfish
```
????

comma commented on maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dm:
```diff
@@ -0,0 +1,60 @@
+
+/obj/effect/shuttle_landmark/hyperion/start
+	name = "Hyperion Crashsite"
+	// landmark_tag = "nav_hangar_bearcat_one"
```
?????????

biscuitWizard replied:
Those aren't real questions.

comma commented on maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion_jobs.dm:
```diff
@@ -0,0 +1,55 @@
+
+/decl/hierarchy/outfit/job/hyperion/pathfinder
+	name = HYPERION_OUTFIT_JOB_NAME("Pathfinder")
+	uniform = /obj/item/clothing/under/casual_pants/classicjeans
```
this seems to be just a copypaste of bearcat captain, down to hawaii shirt equipping.
Pathfinder (C) (TM) is a trademark of SCG Expeditionary Corps, if they're not supposed to be from there just call them Expedition Leader

biscuitWizard replied:
Oh, sure. Do you think jumpsuits would work?

comma replied:
Well you either go full EC (would probably get awkward with includes) or go for a jumpsuit (maybe not grey one so they're not freaking greytides) and rename the job. Pathfinder is a meme title, not a real job.

Commit:
Giving crew members a basic jumpsuit to go with pathfinder title. Changing ionic anomaly to bluespace anomaly. Fixing some access copy-pasta errors.

biscuitWizard commented:
![image](https://user-images.githubusercontent.com/1904063/65406309-0036cb00-dd94-11e9-8a9d-2296c92fc45d.png)

Commit:
Changing network for R-UST components just to be safe.

comma commented:
You also commiteed backup file

comma commented:
re: map

- I don't see an actual cycling airlock there, just sticking a bunch of vents and scrubbers won't save you.
- Can those shutters to RUST core be controlled?  Not seeing a button anywhere.
- Window pages in front of engines aren't needed, engines are dense and air blocking.
- You seem to be missing area in whole center part of the ship, unless it just doesnt have an icon

comma commented on maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion_access.dm:
```diff
@@ -0,0 +1,17 @@
+/var/const/access_hyperion_pathfinder = "ACCESS_HYPERION_PATHFINDER" //997
+/datum/access/hyperion_pathfinder
+	id = access_hyperion_pathfinder
+	desc = "IND Pathfinder"
```
Rename from Pathfinder to Expedition Leader or something other generic like that.
Pathfinder is EC specific job title.

biscuitWizard commented:
> re: map
> 
> * I don't see an actual cycling airlock there, just sticking a bunch of vents and scrubbers won't save you.
Correct. No docking controller is installed. I meant to do so and forgot.
> * Can those shutters to RUST core be controlled?  Not seeing a button anywhere.
They would if I had not accidentally overwritten the button.
> * Window pages in front of engines aren't needed, engines are dense and air blocking.
It's a standard convention across the map.
> * You seem to be missing area in whole center part of the ship, unless it just doesnt have an icon
It just doesn't have an icon, it is area'd.

comma commented:
>It's a standard convention across the map.

It's old thing that no one bothered to fix from when engines failed ot properly block air, not a real convention.

comma commented:
```
maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dmm:482:error: undefined type: /obj/machinery/power/smes/buildable/preset/torch/engine_main
maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dmm:484:error: unknown type
maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dmm:571:error: undefined type: /obj/machinery/suit_storage_unit/explorer
maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dmm:573:error: unknown type
maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dmm:930:error: undefined type: /obj/machinery/suit_storage_unit/explorer
maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dmm:932:error: unknown type
maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dmm:1662:error: undefined type: /obj/machinery/vending/engineering/bearcat
maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dmm:1664:error: unknown type
```
you also used bunch of torch/bearcat specific things on the map it seems

Commit:
Added button for cockpit shutters.
Added button for reactor shutters.
Added embedded controller for the airlock.
Removed engi-vend from bearcat and replaced with YouTool.
Removed the suit storage units since they're Torch somehow (??????)
Renaamed Pathfinder to Expedition Leader.

comma commented on maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion_jobs.dm:
```diff
@@ -0,0 +1,45 @@
+	id_type = /obj/item/weapon/card/id/hyperion
+
+/decl/hierarchy/outfit/job/hyperion/pathfinder
+	name = HYPERION_OUTFIT_JOB_NAME("Pathfinder")
```
re still calling them that in some spots.
IMO repath stuff to so it matches the actual name, would be easier for people to search if needed.

comma commented on maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dm:
```diff
@@ -0,0 +1,45 @@
+	suffixes = list("hyperioncrash/hyperion.dmm")
+	cost = 2
+	template_flags = TEMPLATE_FLAG_CLEAR_CONTENTS | TEMPLATE_FLAG_NO_RUINS | TEMPLATE_FLAG_NO_RADS
+	ruin_tags = RUIN_HUMAN|RUIN_HABITAT
```
Unsure if you want HABITAT flag there - it's for 'ruins' that are long term settlement sorta deals, so it would only spawn on habitable planets.
If you wanted them to spawn on any kind of planet, RUIN_WRECK is one you want, it'll get spawned on any kind of planet that could have ships crashed onto it (currently all IIRC)

comma commented on maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dm:
```diff
@@ -0,0 +1,45 @@
+	cost = 2
+	template_flags = TEMPLATE_FLAG_CLEAR_CONTENTS | TEMPLATE_FLAG_NO_RUINS | TEMPLATE_FLAG_NO_RADS
+	ruin_tags = RUIN_HUMAN|RUIN_HABITAT
+	apc_test_exempt_areas = list(
```
don't need empty list re, it's null by default.

comma commented on maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dm:
```diff
@@ -0,0 +1,45 @@
+	id = "hyperion"
+	description = "a crashed long range shuttle"
+	suffixes = list("hyperioncrash/hyperion.dmm")
+	cost = 2
```
It looks smol enough that it could do with a lower cost, like 1. Cost of 2 is like half of entire default budget.

comma commented on maps/random_ruins/exoplanet_ruins/hyperioncrash/hyperion.dm:
```diff
@@ -0,0 +1,45 @@
+	max_speed = 1/(6 SECONDS)
+	burn_delay = 6 SECONDS
+	fore_dir = NORTH
+	vessel_size = SHIP_SIZE_LARGE
```
fyi this will make it not able to evade meteors on low speed, a-la Charon.

biscuitWizard replied:
Should it be able to, you feel?

comma replied:
It doesn't look that much bigger than Charon, so yeah go for it. Split there is mostly 'would this ship be able to fly through meteors without instantly dying'.

afterthought2 commented:
@biscuitWizard

comma commented:
I guess it's abandoned. Reopen if not.
